pyartcd/pyartcd/pipelines/prepare_release_konflux.py

```diff
@@ -0,0 +1,602 @@
+        return AsyncErrataAPI()
+
+    @cached_property
+    def _gitlab(self) -> gitlab.Gitlab:
+        gl = gitlab.Gitlab(self.gitlab_url, private_token=self.gitlab_token)
+        gl.auth()
+        return gl
+
+    @property
+    def release_name(self) -> str:
+        return get_release_name_for_assembly(self.group, self.releases_config, self.assembly)
+
+    @property
+    def assembly_group_config(self) -> dict:
+        return self.releases_config["releases"][self.assembly].setdefault("assembly", {}).setdefault("group", {})
+
+    @property
+    def shipment_config(self) -> dict:
+        shipment_key = next(k for k in self.assembly_group_config.keys() if k.startswith("shipment"))
+        return self.assembly_group_config.get(shipment_key, [])
+
+    def setup_working_dir(self):
+        self.working_dir.mkdir(parents=True, exist_ok=True)
+        if self._build_repo_dir.exists():
+            shutil.rmtree(self._build_repo_dir, ignore_errors=True)
+        if self._shipment_repo_dir.exists():
+            shutil.rmtree(self._shipment_repo_dir, ignore_errors=True)
+        if self.elliott_working_dir.exists():
+            shutil.rmtree(self.elliott_working_dir, ignore_errors=True)
+
+    async def setup_repos(self):
+        await self.build_data_repo.setup(
+            remote_url=self.build_data_push_url,
+            upstream_remote_url=self.build_repo_pull_url,
+        )
+        await self.build_data_repo.fetch_switch_branch(self.build_data_gitref or self.group)
+        await self.shipment_data_repo.setup(
+            remote_url=self.basic_auth_url(self.shipment_repo_push_url, self.gitlab_token),
+            upstream_remote_url=self.shipment_repo_pull_url,
+        )
+        await self.shipment_data_repo.fetch_switch_branch("main")
+
+        self.releases_config = yaml.load(await self.build_data_repo.read_file("releases.yml"))
+        self.group_config = yaml.load(await self.build_data_repo.read_file("group.yml"))
+
+    async def validate_assembly(self):
+        # validate assembly and init release vars
+        if self.releases_config.get("releases", {}).get(self.assembly) is None:
+            raise ValueError(f"Assembly not found: {self.assembly}")
+        assembly_type = get_assembly_type(self.releases_config, self.assembly)
+        if assembly_type == AssemblyTypes.STREAM:
+            raise ValueError("Preparing a release from a stream assembly is no longer supported.")
+
+        # validate product from group config
+        merged_group_config = assembly_group_config(
+            Model(self.releases_config), self.assembly, Model(self.group_config)
+        ).primitive()
+        group_product = merged_group_config.get("product", self.product)
+        if group_product != self.product:
+            raise ValueError(
+                f"Product mismatch: {group_product} != {self.product}. This pipeline only supports {self.product}."
+            )
+
+    async def run(self):
```

Comment:
Nice! Maybe bump this to the first method after `__init__`, then it really functions as a TOC for human readers.